platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/utils/OfflineUtils.java

```diff
@@ -0,0 +1,41 @@
+package com.mapbox.mapboxsdk.testapp.utils;
+
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import com.mapbox.mapboxsdk.offline.OfflineRegion;
+
+import org.json.JSONObject;
+
+import static com.mapbox.mapboxsdk.testapp.activity.offline.OfflineActivity.JSON_CHARSET;
+import static com.mapbox.mapboxsdk.testapp.activity.offline.OfflineActivity.JSON_FIELD_REGION_NAME;
+
+public class OfflineUtils {
+
+    public static String convertRegionName(@NonNull  byte[] metadata) {
+        String regionName;
+        try {
+            String json = new String(metadata, JSON_CHARSET);
+            JSONObject jsonObject = new JSONObject(json);
+            regionName = jsonObject.getString(JSON_FIELD_REGION_NAME);
+        } catch (Exception exception) {
+            Log.e("OfflineUtils", "Failed to decode metadata: " + exception.getMessage());
+            regionName = "Region x";
```

Comment:
For consistency with `convertRegionName` let's return a `null` here too if this fails and let the `UpdateMetadataActivity` handle it (what you had with `"Region: %d ", OfflineRegion.getId()` was fine as `Region x` will look like the same region is duplicated if several regions fail here).